### PR TITLE
tyr: send error 409 while deleting billing_plan used in user

### PR DESCRIPTION
**Pull request by kadhikari**
1. While trying to delete a billing_plan used by an user (referred in an user), tyr returns an error 500 (Internal server error).
2. API modified to return an error 409 with a message  : {'error': 'billing_plan used'}, 409)

### CODE REVIEW

**Comment by antoine-de** [comment]
can you add a unit test ?

**Review by antoine-de — APPROVED** [review]
:+1: great tests

File: source/tyr/tests/integration/users_test.py [hunk 1]
```diff
@@ -647,3 +656,39 @@ def test_get_users_with_disable_geojson_false(create_multiple_users, geojson_pol
...
+    assert status == 204
+
+
+def test_delete_billing_plan_used_by_an_user(create_user, geojson_polygon):
+    """
+    We start by creating the user with a shape.
+    We request the user with parameter disable_geojson=false
+    We try to delete the billing_plan of this user but in vain.
+    """
+    resp = api_get('/v0/users/{}?disable_geojson=false'.format(create_user))
```
Comment (antoine-de) [inline source/tyr/tests/integration/users_test.py]: I don't see how you added the billing plan here ?

Comment (kadhikari) [reply source/tyr/tests/integration/users_test.py]: The function "create_user" uses a default billing plan

**Comment by antoine-de** [comment]
Jenkins, test this please

**Comment by antoine-de** [comment]
Jenkins, test this please

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**